sys/boot/i386/common/bootargs.h

```diff
@@ -64,10 +64,28 @@ struct bootargs
 	 */
 };
 
+#ifdef LOADER_GELI_SUPPORT
+#include <crypto/intake.h>
+#endif
+
 struct geli_boot_args
 {
     uint32_t		size;
-    char		gelipw[256];
+    union {
+        char            gelipw[256];
+        struct {
+            char                notapw;	/* 
+					 * single null byte to stop keybuf
+					 * being interpreted as a password
+					 */
+            uint32_t            keybuf_sentinel;
```

Comment:
this sentinel is for the opposite case, an older gptzfsboot and a newer zfsloader

If a password is set and we access this member, we would find plain text and try to use it as a key. This magic value lets us ensure what is here is actually a keybuf from boot2